### ISSUE TITLE
Allow passing arbitrary data to fields/widgets through root registry

### ISSUE THREAD

**Issue by olzraiti** [mozilla-services/react-jsonschema-form#278]
We'd like to pass some data to all our custom fields/widgets from the Form root to make some fields/widgets context aware. I think the `registry` object could be utilised for this.

This would make it possible to implement any context aware fields/widgets. This would be useful for example for passing translations (at least for now, since there is no support for translations yet).

Now we pass registry fields and widgets like so:

```
<Form
  fields={fields}
  widgets={widgets}
/>
```

What we'd like to have is this:

```
<Form
  registry={
      fields: fields,
      widgets: widgets,
      someArbitraryObject: someArbitraryObject}
/>
```

**Comment by n1k0** [comment]
This is a good idea, but I'd rather distinguish it from the registry. We could introduce a `context` property we would propagate to each fields and widgets.

**Comment by olzraiti** [comment]
Sounds good. I can write a pull request, since I did this already on our private fork.

**Comment by n1k0** [comment]
@olzraiti that'd be great, thanks!

**Comment by n1k0** [comment]
Released in [v0.40.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.40.0).
